src/Rules/Properties/PropertyInClassRule.php

```diff
@@ -32,6 +32,18 @@ public function processNode(Node $node, Scope $scope): array
 			return [];
 		}
 
+		if (
+			$node->isFinal()
+			&& !$this->phpVersion->supportsFinalProperties()
+		) {
+			return [
+				RuleErrorBuilder::message('Property cannot be final.')
```

Comment:
This needs to follow a message pattern of other features that have support in recent PHP versions. Example: Property hooks are supported only on PHP 8.4 and later.